warehouse/query-core/src/main/java/datawave/query/jexl/visitors/PushdownUnindexedFieldsVisitor.java

```diff
@@ -0,0 +1,171 @@
+        return (T) (queryTree.jjtAccept(visitor, null));
+    }
+
+    @Override
+    public Object visit(ASTAndNode node, Object data) {
+        // if not already delayed somehow
+        if (!QueryPropertyMarker.findInstance(node).isAnyTypeExcept(BOUNDED_RANGE)) {
+            LiteralRange range = JexlASTHelper.findRange().getRange(node);
+
+            if (range != null) {
+                return delayBoundedIndexHole(range, node, data);
+            } else {
+                JexlNode andNode = JexlNodes.newInstanceOfType(node);
+                JexlNodes.copyIdentifierOrLiteral(node, andNode);
+                andNode.jjtSetParent(node.jjtGetParent());
+
+                // We have no bounded range to replace, just proceed as normal
+                JexlNodes.ensureCapacity(andNode, node.jjtGetNumChildren());
+                for (int i = 0; i < node.jjtGetNumChildren(); i++) {
+                    JexlNode newChild = (JexlNode) node.jjtGetChild(i).jjtAccept(this, data);
+                    andNode.jjtAddChild(newChild, i);
+                    newChild.jjtSetParent(andNode);
+                }
+                return andNode;
+            }
+        } else {
+            return node;
+        }
+    }
+
+    /**
+     * Delay the ranges that overlap holes. The range map is expected to only be indexed ranges.
+     *
+     * @param data
+     *            the node data
+     * @param range
+     *            the range
+     * @param currentNode
+     *            the current node
+     * @return a jexl node
+     */
+    protected JexlNode delayBoundedIndexHole(LiteralRange range, ASTAndNode currentNode, Object data) {
+
+        if (missingIndexRange(range)) {
+            return QueryPropertyMarker.create(currentNode, INDEX_HOLE);
+        } else {
+            return currentNode;
+        }
+
+    }
+
+    @Override
+    public Object visit(ASTReferenceExpression node, Object data) {
+        // if not already delayed somehow
+        if (!QueryPropertyMarker.findInstance(node).isAnyTypeExcept(BOUNDED_RANGE)) {
+            return super.visit(node, data);
+        }
+        return node;
+    }
+
+    @Override
+    public Object visit(ASTEQNode node, Object data) {
+        if (isIndexed(node) && missingIndexRange(node)) {
+            return QueryPropertyMarker.create(node, INDEX_HOLE);
+        }
+        return node;
+    }
+
+    @Override
+    public Object visit(ASTERNode node, Object data) {
+        if (isIndexed(node) && missingIndexRange(node)) {
+            return QueryPropertyMarker.create(node, INDEX_HOLE);
+        }
+        return node;
+    }
+
+    public boolean isIndexed(JexlNode node) {
+        String field = JexlASTHelper.getIdentifier(node);
+        try {
+            return (field != null && this.helper.isIndexed(field, this.dataTypeFilter));
+        } catch (TableNotFoundException e) {
+            throw new IllegalStateException("Unable to find metadata table", e);
+        }
+    }
```

Comment:
I question whether the metadata helper is actually necessary for this. You already have a set of non-indexed fields...by definition isn't every field not in that set indexed? If you do need to determine the indexed state of a field, why not remove the metadata helper and simply add the set of indexed fields? Then you can remove the set of non-indexed fields.

Reply:
I will see what I can do here.

Reply:
Actually there is no need to check for indexed fields as the unindexed field list is enough for this visitor..... fixing.